cli/src/commands/bookmark/mod.rs

```diff
@@ -201,3 +203,92 @@ fn is_fast_forward(repo: &dyn Repo, old_target: &RefTarget, new_target_id: &Comm
         true
     }
 }
+
+fn validate_bookmark_names(
+    command: &CommandHelper,
+    ui: &mut Ui,
+    bookmark_names: &[String],
+    command_name: &str,
+) -> Result<(), CommandError> {
+    let helper = command.workspace_helper_no_snapshot(ui)?;
+
+    let mut remote_like = Vec::new();
+    let mut revset_like = Vec::new();
+
+    for name in bookmark_names {
+        match BookMarkKind::parse(name, &helper)? {
+            BookMarkKind::RemoteLike => remote_like.push(name.as_str()),
+            BookMarkKind::RevsetLike => revset_like.push(name.as_str()),
+            BookMarkKind::Normal => {}
+        }
+    }
+
+    if !remote_like.is_empty() {
+        let mut writer = ui.warning_with_heading("Bookmarks containing '@':");
+        for name in &remote_like {
+            writeln!(writer, "  {name}")?;
+        }
+        drop(writer);
+
+        let mut hint = ui.hint_default();
+        writeln!(
+            hint,
+            "To track remote bookmarks, use: jj bookmark --track {}",
+            remote_like.join(" ")
+        )?;
```

Comment:
Unparsable name can be just an error, not a warning. I think that's the idea of using revset parser. User can use escape syntax to specify weird bookmark name.

Reply:
> Unparsable name can be just an error, not a warning. I think that's the idea of using revset parser. User can use escape syntax to specify weird bookmark name.

Sorry, I didn't get it. The only reason to use a revset parser is not to reinvent my own :)
You want to return an error for cases when name was sucessfully parsed as revset(except legit cases) and when it was not parsed?

Reply:
Yes, I think Yuya just meant the since we're using the real revset parser, we can make it an error. If we had some naive check instead, we might have wanted to make it a warning so we don't prevent valid names. But I'm not 100% sure that's what he meant.

Reply:
Yes.

without shell escaping:
* `foo@bar` -> error
* `"foo@bar"` -> local bookmark `foo@bar`
* `foo-` -> error
* `"foo-"` -> local bookmark `foo-`
* `(foo)` -> could be an error (but might be okay to parse as local bookmark `foo`)